packages/react-components/react-positioning/src/usePositioning.ts

```diff
@@ -111,38 +70,19 @@ export function usePositioning(
           console.warn(err.stack);
         }
 
-        overrideTargetRef.current = target;
+        setOverrideTarget(target);
       },
     }),
-    // Missing deps:
-    // options.target - only used for a runtime warning
-    // overrideTargetRef - Stable between renders
-    // updatePosition - Stable between renders
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    [],
+    [options.target, setOverrideTarget],
   );
 
   useIsomorphicLayoutEffect(() => {
-    overrideTargetRef.current = options.target ?? null;
-  }, [options.target, overrideTargetRef, containerRef]);
-
-  useIsomorphicLayoutEffect(() => {
-    updatePosition();
-  }, [enabled, resolvePositioningOptions, updatePosition]);
+    setOverrideTarget(options.target ?? null);
+  }, [options.target, setOverrideTarget]);
 
-  // Add window resize and scroll listeners to update position
   useIsomorphicLayoutEffect(() => {
-    const win = targetDocument?.defaultView;
-    if (win) {
-      win.addEventListener('resize', updatePosition);
-      win.addEventListener('scroll', updatePosition);
-
-      return () => {
-        win.removeEventListener('resize', updatePosition);
-        win.removeEventListener('scroll', updatePosition);
-      };
-    }
-  }, [updatePosition, targetDocument]);
+    updatePositionManager();
+  }, [enabled, resolvePositioningOptions, updatePositionManager]);
```

Comment:
```suggestion
  }, [updatePositionManager]);
```

Redundant dependencies

Reply:
no, we want these dependencies because we want position to update after rerender when:
* enabled changes from false -> true
* options change

Reply:
We don't 😜 `updatePositionManager` already has `enabled` & `resolvePositioningOptions` as dependencies.

Reply:
ah yeah it's not an event callback anymore